binance-pipeline/lambdas/binance/binance_utils.py

```diff
@@ -0,0 +1,82 @@
+# ./lambdas/binance/binance_utils.py
+from datetime import datetime, timedelta
+import itertools
+import typing
+#
+from binance.binance_api import BinanceQueryAPI, get_latest_trade_id, get_first_trade_id
+
+def get_date(payload: dict):
+    time_ms = payload['time']
+    time_s = time_ms*1e-3
+
+    return datetime.fromtimestamp(time_s)
+
+
+def get_trade_id(payload: dict):
+
+    return payload['id']
+
+
+class TradeIDFinder:
+    def __init__(self, symbol, start_date: typing.Optional[datetime]=None):
+        self.symbol = symbol
+        if start_date:
+            self.start_date = start_date - timedelta(days=1)
+        else:
+            self.start_date = None
+        self.query_api = BinanceQueryAPI()
+
+        return
+
+    def make_bounds(self, first_trade_payload: dict, last_trade_payload: dict):
+        bounds = dict()
+        for payload in [first_trade_payload, last_trade_payload]:
+            bounds[get_trade_id(payload)] = get_date(payload)
+
+        return bounds
+
+    def get_fraction_from_days(self, first_date, last_date):
+        days_before = (self.start_date - first_date).days
```

Comment:
Can this be negative? Worth adding a check?

Reply:
I will add it to the unit tests which are currently low in coverage but from memory I think this is not possible, by design it should be impossible anyway